resteasy/src/main/java/com/linecorp/armeria/server/resteasy/SecurityContextImpl.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.resteasy;
+
+import static java.util.Objects.requireNonNull;
+
+import java.security.Principal;
+
+import javax.annotation.Nullable;
+import javax.ws.rs.core.SecurityContext;
+
+import org.jboss.resteasy.plugins.server.embedded.SecurityDomain;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * Implements {@link SecurityContext}.
+ */
+@UnstableApi
+final class SecurityContextImpl implements SecurityContext {
+
+    public static SecurityContext basic(Principal principal, SecurityDomain securityDomain) {
+        return new SecurityContextImpl(principal, securityDomain, "BASIC", true);
+    }
+
+    public static SecurityContext unsecure() {
```

Comment:
`insecure` for consistency?
ClientFatory already uses it.
https://github.com/line/armeria/blob/3d30b065ee1a1542f54ffe3a03dea6b4cc36c959/core/src/main/java/com/linecorp/armeria/client/ClientFactory.java#L84

Reply:
Sure. Renamed